corehq/apps/api/domain_metadata.py

```diff
@@ -62,7 +76,16 @@ def obj_get_list(self, bundle, **kwargs):
         if kwargs.get('domain'):
             return [self.obj_get(bundle, **kwargs)]
         else:
-            return list(Domain.get_all())
+            filters = {}
+            if hasattr(bundle.request, 'GET'):
+                filters = bundle.request.GET
+            domains = list(Domain.get_all())
```

Comment:
This is a reasonable approach, but looking at our code it seems like we already have this data in the "hqdomains" elasticsearch index. I just added the `last_modified` field to the index (see https://github.com/dimagi/commcare-hq/pull/8152) so that you can use it this way. Can you try a similar approach but using this elasticsearch index?